app/containers/TrackPopupContainer/index.js

```diff
@@ -23,13 +23,33 @@ const TrackPopupContainer = props => {
     actions,
     musicSources,
     settings,
+    doubleClick,
 
     withAddToQueue,
     withPlayNow,
     withAddToFavorites,
     withAddToDownloads
   } = props;
-  
+
+  const playSong = () => {
+    if (withPlayNow) {
+      actions.clearQueue();
+      actions.addToQueue(
+        musicSources,
+        {
+          artist: props.artist,
+          name: props.title,
+          thumbnail: props.thumb
+        });
+      actions.selectSong(0);
+      actions.startPlayback();
+    }
+  };
+
+  if (doubleClick) {
+    playSong();
```

Comment:
Anything that dispatches actions should stay out of the render method, and only do so in response to some event.